sklearn/feature_extraction/hashing.py

```diff
@@ -148,9 +158,19 @@ def transform(self, raw_X):
             raw_X = (_iteritems(d) for d in raw_X)
         elif self.input_type == "string":
             raw_X = (((f, 1) for f in x) for x in raw_X)
+        # expose the seed for HashingVectorizer
+        if isinstance(self.random_state, int):
```

Comment:
I don't think that it will work with `np.int32` or another integer type.

```python
from numbers import Integral
```

```suggestion
        if isinstance(self.random_state, Integral):
```